apm-agent-api/src/main/java/co/elastic/apm/api/AbstractSpanImpl.java

```diff
@@ -151,4 +151,40 @@ public void injectTraceHeaders(HeaderInjector headerInjector) {
     private void doInjectTraceHeaders(MethodHandle addHeader, HeaderInjector headerInjector) {
         // co.elastic.apm.agent.plugin.api.AbstractSpanInstrumentation.InjectTraceHeadersInstrumentation
     }
+
+    @Nonnull
+    @Override
+    public Span withDestinationServiceResource(String resource) {
```

Comment:
Hm, I think we're hitting some limits with how we modelled spans and transactions here. These methods are only relevant for spans, not transactions. But in the public API, transaction extends span. We should think about introducing a `BaseSpan` that contains methods that are relevant for both spans and transactions. But that would likely break binary compatibility.

Reply:
hi @felixbarny , please review changes